### PR TITLE
ZK-4079: progressmeter animation not catching up with actual value

### PR DESCRIPTION
**Pull request by plijyfes**

### CODE REVIEW

File: zul/src/archive/web/js/zul/wgt/Progressmeter.js [hunk 1]
```diff
@@ -43,8 +43,8 @@ zul.wgt.Progressmeter = zk.$extends(zul.Widget, {
 			if (zk(n).isRealVisible()) { //Bug 3134159
 				var $img = jq(img);
 				$img.animate({
-					width: Math.round((jq(n).innerWidth() * this._value) / 100) + 'px'
-				}, $img.zk.getAnimationSpeed('slow'));
+					width: this._value + '%'
+				}, { duration: $img.zk.getAnimationSpeed('slow'), queue: false, easing: "linear" }); //ZK-4079: progressmeter animation not catching up with actual value
```
Comment (charlesqqq) [inline zul/src/archive/web/js/zul/wgt/Progressmeter.js]: 'linear'

File: zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4079Test.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		click(jq("$veryhigh"));
+		waitResponse();
+		Assert.assertTrue(getBarWidth() >= 800);
+	}
+	
+	private Integer getBarWidth() {
+		WebDriverWait wait = new WebDriverWait(driver, 7);
+		wait.until(ExpectedConditions.alertIsPresent());
+		Alert alert = driver.switchTo().alert();
+		Integer barWidth = Integer.valueOf(alert.getText().substring(alert.getText().length()-3));
```
Comment (charlesqqq) [inline zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4079Test.java]: length() - 3

File: zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4079Test.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		click(jq("$veryhigh"));
+		waitResponse();
+		Assert.assertTrue(getBarWidth() >= 800);
+	}
+	
+	private Integer getBarWidth() {
+		WebDriverWait wait = new WebDriverWait(driver, 7);
+		wait.until(ExpectedConditions.alertIsPresent());
+		Alert alert = driver.switchTo().alert();
+		Integer barWidth = Integer.valueOf(alert.getText().substring(alert.getText().length() - 3));
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/B86_ZK_4079Test.java]: Alert text pattern `100% and the bar width is: XXX`

Then extract string by `split(": ")[1]`

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**